### PR TITLE
Fix (AddressBook): Accept BCH addresses

### PR DESCRIPTION
**Pull request by OjusWiZard**
Closes https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=51733029

## Checklist

- [x] Validate BCH addresses when adding them to address book
- [x] Also, move `is_valid_base58_address` to `...bch/utils.py` to avoid circular import
- [x] Update a test to check validation

### CODE REVIEW

File: rotkehlchen/chain/bitcoin/utils.py [hunk 1]
```diff
@@ -39,7 +43,8 @@ def is_valid_btc_address(value: str) -> bool:
     return (
         is_valid_base58_address(value) or
         is_valid_bech32_address(value) or
-        is_valid_bech32_bip350_address(value)
+        is_valid_bech32_bip350_address(value) or
+        is_valid_bitcoin_cash_address(value)
```
Comment (yabirgb) [inline rotkehlchen/chain/bitcoin/utils.py:47]: is this true? all the valid bch addresses are valid btc addresses? The set of addresses is not different between both of them (until the split sure it is valid)

Comment (OjusWiZard) [reply rotkehlchen/chain/bitcoin/utils.py]: right, they are not the same

**Review by yabirgb — APPROVED** [review]
Failing test failed due to Remote error:
```
DEBUG    rotkehlchen.api.server:server.py:444 Greenlet-405: end rotki api PUT /api/1/blockchains/BCH/accounts view_args={'blockchain': 'BCH'}, query_string=b'', status_code=400, result={'message': '{"address": ["Given ENS address prettyirrelevant.eth could not be resolved for bitcoin cash due to: Please check your network and confirm sufficient nodes are connected for ethereum."]}', 'result': None}
```
passes locally

File: rotkehlchen/api/v1/schemas.py [hunk 2]
```diff
@@ -2611,7 +2618,8 @@ def validate_schema(
         supported_blockchain_type = blockchain.get_chain_type()
         if ((
             supported_blockchain_type == 'bitcoin' and
-            is_valid_btc_address(data['address']) is False
+            is_valid_btc_address(data['address']) is False and
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:2621]: shouldn't this better check the exact value and do `is_valid_btc_address` for `SupportedBlockchain.BITCOIN` and `is_valid_bitcoin_cash_address` for `SupportedBlockchain.BITCOIN_CASH` ?

**Review by LefterisJP — COMMENTED** [review]
Needs changes and also related test is failing: `test_add_blockchain_accounts`

**Comment by OjusWiZard** [comment]
2 failed tests (api) are due to [unsynced cassettes](https://github.com/rotki/rotki/actions/runs/7817571695/job/21325925863?pr=7409#step:10:2207)
1 failed test (other) is due to [Etherscan read timeout](https://github.com/rotki/rotki/actions/runs/7817571695/job/21325926201?pr=7409#step:10:493)

**Review by LefterisJP — APPROVED** [review]
lgtm

**Pull request merged**